OpenSeadragon v6 support

Aiosa opened the pull request:
Apart from OSD v6 support, it enables:
 - API-fied tiff handling
 - worker processing
 - better dev mode: live reload, automated opening
 
Fixed bad api usages. Removed monkeypatching -> you are using `downloadTileStart` which was introduced in OSD v4.1 (by me). The patch is applied for pre-v4 versions, which won't work anyway.

Code review is welcome, I did not have much time to check it, mostly it was done by a chatbot.

My motivation is being able to feed multichannel data directly to a drawer, which will be possible in OSD v6.

pearcetm commented:
> Also, we would not put it into the odd repository directly, we would move it to the OpenSeadragon project as a new repository, similar to the flex-renderer.

Oh ok, that makes more sense. I'd be fine with that.

iangilman commented:
Cool... I'm fine with moving it into the OSD GitHub organization. Should we use the same name? @pearcetm I would expect you to still be the maintainer of the repository... does that make sense?

Aiosa commented:
I added tons of changes. The summary:

- fixed tests - running in dev mode on real files, not dist/ output -> you can code and watch reload tests
- added tests - correct parsing of tiff files is tested
- tests now run on playwright - testing workers is hard to do otherwise
- added support for planar bands tiff encoding
  - detects if image like or data like payload  
  - it can now parse also mutlichannel data and, if forced (~image conversion), falls back to 4chan + warning 
- added support for better pyramid resolution
  - see discussion above, it is ready for SubIFd if the underlying library ever adds support, otherwise it shows a warning  
- added options file that can control the default behavior (e.g. whether to treat as image or as a data payload, other parsing setup)
- added direct tiff rendering support via publishing new types
   -> you don't have a tiff file, but server delivers tiff tiles
   -> your renderer supports rendering more than 4 channels and you avoid image conversion (implemented and soon published on flex-rednerer).

Needs fix:
 - osd v6 is hardcoded everywhere and NPM avoided as it was not published at the time of writing

Possible improvements:
 - webassembly

What we need:
 - can you guys test this, if it does not break your stuff? The demos still work for me, and the renderer too, though I need to now test also more than 4 channels once I find some file that actually has them

<img width="1872" height="792" alt="image" src="https://github.com/user-attachments/assets/920b611c-4066-4b10-b5e2-cfcd7de0b635" />

Aiosa commented:
I think I am done. If you need something more...

pearcetm commented:
@Ramxnchv Yes, I just created https://github.com/pearcetm/GeoTIFFTileSource/releases/tag/v2.3.0
